actions/generate_changelog.go

```diff
@@ -0,0 +1,144 @@
+package actions
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+	"sync"
+	"text/template"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+const (
+	changelogTpl string = `{{.OldRelease}} -> {{.NewRelease}}
+
+# Features
+
+{{range .Features}} - {{.}}
+{{else}}No new features for this release.
+{{end}}
+
+# Fixes
+
+{{range .Fixes}} - {{.}}
+{{else}}No bug fixes for this release.
+{{end}}
+
+# Documentation
+
+{{range .Documentation}} - {{.}}
+{{else}}No new documentation for this release.
+{{end}}
+
+# Maintenance
+
+{{range .Maintenance}} - {{.}}
+{{else}}No maintenance required for this release.
+{{end}}`
+)
+
+type Changelog struct {
+	OldRelease string
+	NewRelease string
+	Features []string
+	Fixes []string
+	Documentation []string
+	Maintenance []string
+}
+
+// GenerateChangelog is the CLI action for creating an aggregated changelog from all of the Deis Workflow repos.
+func GenerateChangelog(client *github.Client, dest io.Writer) func(*cli.Context) {
+	return func(c *cli.Context) {
+		changelog := &Changelog{
+			OldRelease: c.Args().Get(0),
+			NewRelease: c.Args().Get(1),
+		}
+		if changelog.OldRelease == "" || changelog.NewRelease == "" {
+			log.Fatal("Usage: generate-changelog <old-release> <new-release>")
+		}
+		if err := generateChangelog(client, changelog); err != nil {
+			log.Fatalf("could not generate changelog: %s", err)
+		}
+		t := template.Must(template.New("changelog").Parse(changelogTpl))
```

Comment:
@bacongobbler can you put this `template.Must` call in global scope, so it gets executed before anything else? more of a defensive programming measure than anything else.